pkg/server/plugin/nodeattestor/aws/iid_test.go

```diff
@@ -497,35 +432,29 @@ func (s *IIDAttestorSuite) TestClientAndIDReturns() {
 				configStr += "\ndisable_instance_profile_selectors = true"
 			}
 
-			_, err := s.p.Configure(context.Background(), &plugin.ConfigureRequest{
-				Configuration: configStr,
-				GlobalConfig:  &plugin.ConfigureRequest_GlobalConfig{TrustDomain: "example.org"},
-			})
-			s.Require().NoError(err)
+			plugin, attestor := s.loadPlugin(clients, plugintest.Configure(configStr),
+				plugintest.HostServices(agentstorev0.AgentStoreServiceServer(s.agentStore)),
+				plugintest.CoreConfig(catalog.CoreConfig{
+					TrustDomain: spiffeid.RequireTrustDomainFromString("example.org"),
+				}))
 
-			data := &common.AttestationData{
-				Type: caws.PluginName,
-				Data: s.iidAttestationDataToBytes(*s.buildDefaultIIDAttestationData()),
-			}
+			payload := s.iidAttestationDataToBytes(*s.buildDefaultIIDAttestationData())
 
 			// using our own keypair (since we don't have AWS private key)
-			originalAWSPublicKey := s.plugin.config.awsCaCertPublicKey
+			originalAWSPublicKey := plugin.config.awsCaCertPublicKey
 			defer func() {
-				s.plugin.config.awsCaCertPublicKey = originalAWSPublicKey
+				plugin.config.awsCaCertPublicKey = originalAWSPublicKey
 			}()
-			s.plugin.config.awsCaCertPublicKey = &s.rsaKey.PublicKey
-
-			resp, err := s.attest(&nodeattestorv0.AttestRequest{
-				AttestationData: data,
-			})
+			plugin.config.awsCaCertPublicKey = &s.rsaKey.PublicKey
 
-			if tt.expectErr != "" {
+			resp, err := attestor.Attest(context.Background(), payload, expectNoChallenge)
+			if tt.expectErrMessage != "" {
 				s.Nil(resp)
-				s.RequireErrorContains(err, tt.expectErr)
+				s.AssertGRPCStatusContains(err, tt.expectErrCode, tt.expectErrMessage)
```

Comment:
Doesn't look like we're asserting that there was no error when tt.expectErrMessage == "". Maybe we can move the call to `AssertGRPCStatusContains` to right after Attest returns?